### PR TITLE
Consume API from focus app

### PR DESCRIPTION
**Pull request by iggy808**
# PROBLEM
Maui application and API are not yet communicating.

# SOLUTION
Implemented Refit within the MAUI app, and configured the android emulator to allow HTTP traffic from the application to the API.

# HOW WAS THIS TESTED?
 With blood sweat and tears. This issue was tested manually by running both the API and MAUI app in debug, and ensuring the request from the MAUI app to the API returns a UserModel.

# DOCUMENTATION
[Link to wiki entry for this issue.](https://github.com/ChrispyPeaches/FocusFriends/wiki/Consuming-API-from-the-MAUI-Application)

### CODE REVIEW

File: src/FocusApp/MauiProgram.cs [hunk 1]
```diff
@@ -23,6 +24,8 @@ public static MauiApp CreateMauiApp()
                     fonts.AddFont("Font-Awesome-6-Free-Regular.otf", nameof(LineArtIcons));
                 });
 
+            builder.Services.AddSingleton<APIService>();
```
Comment (ChrispyPeaches) [inline src/FocusApp/MauiProgram.cs:27]: I think refit's dependency injection might handle the singleton situation. Was this necessary to get the code to work or would we be able to remove it?

**Review by ChrispyPeaches — CHANGES_REQUESTED** [review]
Other than the one question mentioned, it looks good! Nice work overall!

**Review by wesleypayton — DISMISSED** [review]
LGTM!!! 💯

**Review by ChrispyPeaches — APPROVED** [review]
Gettin that money

**Review by wesleypayton — APPROVED** [review]
Still good, now better 💯

**Comment by iggy808** [comment]
Resolved conflicts, refactored implementation to call api upon page load for:

- TimerPage
- SettingsPage
- ShopPage

This is not an optimal implementation- this only serves to demonstrate the fact the API can communicate with the app in a widely integrated way. 

Note: The API calls work when going from the settings page to the timer page- this proves that the instantiated TimerPage holds on to the API client, and allows for the api client to persist across navigation.

**Review by wesleypayton — APPROVED** [review]
LGTM Part 3

**Review by ChrispyPeaches — APPROVED** [review]
💸💸

**Pull request merged**